src/addons/retry.rs

```diff
@@ -0,0 +1,70 @@
+use bevy::prelude::*;
+
+use crate::{systems::spawn_process, LocalCommand, Process, ProcessCompleted};
+
+#[derive(Debug, Component)]
+pub enum Retry {
+    Attempts(usize),
+}
+
+#[derive(Debug, Event)]
+pub struct RetryEvent {
+    pub entity: Entity,
+    pub retries_left: usize,
+}
+
+/// Retry failed processes according to the Retry component.
+///
+/// Processes without the Retry component are ignored.
+/// The Retry component is removed from the entity when retries are done.
+pub(crate) fn retry_failed_process(
+    mut commands: Commands,
+    mut query: Query<(&mut LocalCommand, &mut Process, &mut Retry)>,
+    mut retry_events: EventWriter<RetryEvent>,
+    mut process_completed_events: EventReader<ProcessCompleted>,
+) {
+    for process_completed_event in process_completed_events.read() {
+        if process_completed_event.exit_status.success() {
+            println!("Success, skipping");
+            continue;
+        }
+        if let Ok((mut local_command, mut process, retry)) =
+            query.get_mut(process_completed_event.entity)
+        {
+            match *retry {
+                Retry::Attempts(retries) => {
+                    if let Some(mut entity_commands) =
+                        commands.get_entity(process_completed_event.entity)
+                    {
+                        if retries < 1 {
+                            entity_commands.remove::<Retry>();
+                            continue;
+                        }
+                        let retries_left = retries - 1;
+
+                        // Update the retry attempts
+                        entity_commands.insert(Retry::Attempts(retries_left));
```

Comment:
I'd prefer if we can somehow edit this in-place, maybe adding a `mut` to `retries` and then `retries -= 1`?
Not sure if this is possible, otherwise we can keep it like this and merge

Reply:
I was unable to get this working. It's almost certainly possible but I was not able to figure it out. I did get it compiling with `mut retries` and `retries -= 1`, but the value inside the component never changes - there's a copy happening somewhere that's not obvious to me.

Give it a shot if you like.

Reply:
I did get it working after all.